src/btree/bt_curprev.c

```diff
@@ -553,6 +555,16 @@ __cursor_row_prev(
                 ++*skippedp;
                 continue;
             }
+            /*
+             * If the cursor has prefix search configured we can early exit here if the key we are
+             * visiting is before our prefix.
+             */
+            if (prefix_used && __wt_prefix_match(prefix, key) > 0) {
+                /* It is not okay for the user to have a custom collator. */
+                WT_ASSERT(session, CUR2BT(cbt)->collator == NULL);
```

Comment:
Don't we check this in the reconfigure path?

Reply:
yes we do, I was wondering if we should remove it for the other case too? On line 601 of this file